pkg/multiTenancyPlugins/utils/utils.go

```diff
@@ -92,57 +92,85 @@ func RandStringBytesRmndr(n int) string {
+	//SKIP ...
+	CONTAINERS_PS     commandEnum = "ps"
+	CONTAINERS_JSON   commandEnum = "json"
+	CONTAINER_ARCHIVE commandEnum = "containerArchive"
+	CONTAINER_EXPORT  commandEnum = "containerExport"
+	CONTAINER_CHANGES commandEnum = "containerChanges"
+	CONTAINER_JSON    commandEnum = "containerJson"
+	CONTAINER_TOP     commandEnum = "containerTop"
+	CONTAINER_LOGS    commandEnum = "containerLogs"
+	CONTAINER_STATS   commandEnum = "containerStats"
+	//SKIP ...
+	NETWORKS_LIST   commandEnum = "NetworksList"
+	NETWORK_INSPECT commandEnum = "NetworkInspect"
+	//SKIP ...
+	//POST
+	CONTAINER_CREATE  commandEnum = "containerCreate"
+	CONTAINER_KILL    commandEnum = "containerKill"
+	CONTAINER_PAUSE   commandEnum = "containerPause"
+	CONTAINER_UNPAUSE commandEnum = "containerUnpause"
+	CONTAINER_RENAME  commandEnum = "containerRename"
+	CONTAINER_RESTART commandEnum = "containerRestart"
+	CONTAINER_START   commandEnum = "containerStart"
+	CONTAINER_STOP    commandEnum = "containerStop"
+	CONTAINER_UPDATE  commandEnum = "containerUpdate"
+	CONTAINER_WAIT    commandEnum = "containerWait"
+	CONTAINER_RESIZE  commandEnum = "containerResize"
+	CONTAINER_ATTACH  commandEnum = "containerAttach"
+	CONTAINER_COPY    commandEnum = "containerCopy"
+	CONTAINER_EXEC    commandEnum = "containerExec"
+	//SKIP ...
+
+	CONTAINER_DELETE commandEnum = "containerDelete"
+)
 
-//TODO - Do the same for networks, images, and so on. What is not supported will fail because of the generic message will go to default
+var containersRegexp = regexp.MustCompile("/containers/(.*)/(.*)|/containers/(\\w+)")
+var networksRegexp = regexp.MustCompile("/networks/(.*)/(.*)|/networks/(\\w+)")
+var clusterRegExp = regexp.MustCompile("/(.*)/(.*)")
 
-//TODO - Handle delete better
 func commandParser(r *http.Request) string {
+	containersParams := containersRegexp.FindStringSubmatch(r.URL.Path)
+	networksParams := networksRegexp.FindStringSubmatch(r.URL.Path)
+	clusterParams := clusterRegExp.FindStringSubmatch(r.URL.Path)
 
-	paramsArr1 := containers.FindStringSubmatch(r.URL.Path)
-	paramsArr2 := containersWithIdentifier.FindStringSubmatch(r.URL.Path)
-	networksParams := networks.FindStringSubmatch(r.URL.Path)
-	//assert the it is not possible for two of them to co-Exist
-
-	log.Debug(paramsArr1)
-	log.Debug(paramsArr2)
+	log.Debug(containersParams)
+	log.Debug(networksParams)
+	log.Debug(clusterParams)
 
 	switch r.Method {
 	case "DELETE":
-		if len(paramsArr1) > 0 && strings.HasPrefix(paramsArr1[0], "/containers") {
+		if len(containersParams) > 0 {
 			return "containerdelete"
 		}
-	}
-	//Order IS important
-	if len(paramsArr2) == 3 {
-
-		return "container" + paramsArr2[2]
-	}
-	if len(paramsArr1) == 2 {
+		if len(networksParams) > 0 {
+			return "networkdelete"
+		}
 
-		if paramsArr1[1] == "json" {
-			return "listContainers"
+	case "GET", "POST":
+		if len(containersParams) == 4 && containersParams[2] != "" {
+			return "container" + containersParams[2]
```

Comment:
The enum is defined with capital letter (e.g., containerStart) , I think that the argument in the http request is with lower case ...